### PR TITLE
Remove failing test for FIO38-C

### PR DESCRIPTION
**Pull request by mbaluda**
## Description

- Removes a test for FIO38-C that does compile properly with `qcc`

## Change request type

 - [ ] Release or process automation (GitHub workflows, internal scripts)
 - [ ] Internal documentation
 - [ ] External documentation
 - [x] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [x] No rules added
 - [ ] Queries have been added for the following rules:
     - _rule number here_
 - [ ] Queries have been modified for the following rules:
     - _rule number here_

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [ ] Yes
  - [x] No

🚨🚨🚨
_**Reviewer:**_ Confirm that format of *shared* queries (not the .qll file, the
.ql file that imports it) is valid by running them within VS Code. 
  - [ ] Confirmed

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [ ] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### Reviewer 

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### CODE REVIEW

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_PASS            : True
PACKAGE              : IO2
RULE                 : FIO38-C
COMPILE_PASS         : True
SUITE                : CERT-C
QUERY                : DoNotCopyAFileObject
TEST_DIFFERENCE      : 
COMPILE_ERROR_OUTPUT : 


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_DIFFERENCE      : 
COMPILE_PASS         : False
TEST_PASS            : False
SUITE                : CERT-C
PACKAGE              : IO2
RULE                 : FIO38-C
COMPILE_ERROR_OUTPUT : [2023-03-15 20:14:18] [build-stderr] /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/test.c: In function 'f2b':
                       [2023-03-15 20:14:18] [build-stderr] /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/test.c:16:15: error: lvalue required as unary '&' operand
                       [2023-03-15 20:14:18] [build-stderr]    my_stdout = &stdout;           // COMPLIANT
                       [2023-03-15 20:14:18] [build-stderr]                ^
                       [2023-03-15 20:14:19] [ERROR] Spawned process exited abnormally (code 1; tried to run: [/__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql/tools/linux64/preload_tracer, qcc, -V8.3.0, -Wc,-fsyntax-only, -c, -nopipe, -std=c11, -Vgcc_ntoaarch64le, /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/test.c])
                       
QUERY                : DoNotCopyAFileObject


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

SUITE                : CERT-C
PACKAGE              : IO2
QUERY                : DoNotCopyAFileObject
RULE                 : FIO38-C
COMPILE_ERROR_OUTPUT : 
COMPILE_PASS         : True
TEST_PASS            : True
TEST_DIFFERENCE      : 


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_PASS            : True
PACKAGE              : IO2
COMPILE_PASS         : True
TEST_DIFFERENCE      : 
QUERY                : DoNotCopyAFileObject
COMPILE_ERROR_OUTPUT : 
RULE                 : FIO38-C
SUITE                : CERT-C


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_DIFFERENCE      : 
TEST_PASS            : True
COMPILE_PASS         : True
COMPILE_ERROR_OUTPUT : 
RULE                 : FIO38-C
QUERY                : DoNotCopyAFileObject
SUITE                : CERT-C
PACKAGE              : IO2


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

File: c/cert/test/rules/FIO38-C/test.c.qcc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+int f2(void) {
+  FILE *my_stdout;
+  my_stdout = stdout;           // COMPLIANT
+  FILE my_stdout2 = *my_stdout; // NON_COMPLIANT
+  return fputs("Hello, World!\n", my_stdout);
+}
+int f2b(void) {
+  FILE *const *my_stdout;
+  // my_stdout = &stdout;        // does not compile in qcc
```
Comment (lcartey) [inline c/cert/test/rules/FIO38-C/test.c.qcc:23]: Is this legal C? Or is it that case that you are not guaranteed to be able to take the address of `stdout`?

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

RULE                 : FIO38-C
COMPILE_PASS         : True
TEST_DIFFERENCE      : --- expected
                       +++ actual
                       @@ -1,5 +1,4 @@
                        | test.c:4:20:4:26 | * ... | A FILE object is being copied. |
                        | test.c:11:21:11:30 | * ... | A FILE object is being copied. |
                       -| test.c:17:21:17:31 | * ... | A FILE object is being copied. |
                       -| test.c:23:15:23:21 | * ... | A FILE object is being copied. |
                       -| test.c:36:19:36:28 | * ... | A FILE object is being copied. |
                       +| test.c:17:15:17:21 | * ... | A FILE object is being copied. |
                       +| test.c:30:19:30:28 | * ... | A FILE object is being copied. |
                       [1/1 comp 31.3s eval 1.2s] FAILED(RESULT) /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/DoNotCopyAFileObject.qlref
                       
COMPILE_ERROR_OUTPUT : 
TEST_PASS            : False
SUITE                : CERT-C
PACKAGE              : IO2
QUERY                : DoNotCopyAFileObject


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR won't happen because it is outside of license window!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR won't happen because it is outside of license window!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

QUERY                : DoNotCopyAFileObject
TEST_PASS            : False
RULE                 : FIO38-C
TEST_DIFFERENCE      : --- expected
                       +++ actual
                       @@ -1,5 +1,4 @@
                        | test.c:4:20:4:26 | * ... | A FILE object is being copied. |
                        | test.c:11:21:11:30 | * ... | A FILE object is being copied. |
                       -| test.c:17:21:17:31 | * ... | A FILE object is being copied. |
                       -| test.c:23:15:23:21 | * ... | A FILE object is being copied. |
                       -| test.c:36:19:36:28 | * ... | A FILE object is being copied. |
                       +| test.c:17:15:17:21 | * ... | A FILE object is being copied. |
                       +| test.c:30:19:30:28 | * ... | A FILE object is being copied. |
                       [1/1 comp 30.1s eval 1.2s] FAILED(RESULT) /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/DoNotCopyAFileObject.qlref
                       
PACKAGE              : IO2
COMPILE_PASS         : True
SUITE                : CERT-C
COMPILE_ERROR_OUTPUT : 


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

COMPILE_PASS         : True
SUITE                : CERT-C
QUERY                : DoNotCopyAFileObject
TEST_DIFFERENCE      : --- expected
                       +++ actual
                       @@ -1,5 +1,4 @@
                        | test.c:4:20:4:26 | * ... | A FILE object is being copied. |
                        | test.c:11:21:11:30 | * ... | A FILE object is being copied. |
                       -| test.c:17:21:17:31 | * ... | A FILE object is being copied. |
                       -| test.c:23:15:23:21 | * ... | A FILE object is being copied. |
                       -| test.c:36:19:36:28 | * ... | A FILE object is being copied. |
                       +| test.c:17:15:17:21 | * ... | A FILE object is being copied. |
                       +| test.c:30:19:30:28 | * ... | A FILE object is being copied. |
                       [1/1 comp 32s eval 1.2s] FAILED(RESULT) /__w/codeql-coding-standards-release-engineering/codeql-coding-standards-release-engineering/codeql-coding-standards/c/cert/test/rules/FIO38-C/DoNotCopyAFileObject.qlref
                       
RULE                 : FIO38-C
TEST_PASS            : False
COMPILE_ERROR_OUTPUT : 
PACKAGE              : IO2


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR won't happen because it is outside of license window!

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR won't happen because it is outside of license window!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

RULE                 : FIO38-C
SUITE                : CERT-C
COMPILE_PASS         : True
TEST_DIFFERENCE      : 
COMPILE_ERROR_OUTPUT : 
QUERY                : DoNotCopyAFileObject
TEST_PASS            : True
PACKAGE              : IO2


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

COMPILE_ERROR_OUTPUT : 
SUITE                : CERT-C
RULE                 : FIO38-C
COMPILE_PASS         : True
QUERY                : DoNotCopyAFileObject
PACKAGE              : IO2
TEST_DIFFERENCE      : 
TEST_PASS            : True


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

COMPILE_PASS         : True
PACKAGE              : IO2
QUERY                : DoNotCopyAFileObject
COMPILE_ERROR_OUTPUT : 
TEST_PASS            : True
RULE                 : FIO38-C
TEST_DIFFERENCE      : 
SUITE                : CERT-C


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **clang/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/cpp/X86_64** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **gcc/c/X86_64** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

TEST_DIFFERENCE      : 
QUERY                : DoNotCopyAFileObject
TEST_PASS            : True
SUITE                : CERT-C
COMPILE_PASS         : True
PACKAGE              : IO2
RULE                 : FIO38-C
COMPILE_ERROR_OUTPUT : 


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/cpp/AARCH64LE** Matrix Testing for this PR has been completed but I didn't find anything to test!

**Comment by jsinglet** [comment]
🤖 Beep Boop! **qcc/c/AARCH64LE** Matrix Testing for this PR has been completed. See below for the results! <br><br>
``` 

COMPILE_PASS         : True
SUITE                : CERT-C
PACKAGE              : IO2
TEST_PASS            : True
RULE                 : FIO38-C
COMPILE_ERROR_OUTPUT : 
TEST_DIFFERENCE      : 
QUERY                : DoNotCopyAFileObject


```

**Comment by jsinglet** [comment]
🤖 Beep Boop! Matrix Testing for this PR has been **completed**. If no reports were posted it means this PR does not contain things that need matrix testing!

**Review by jsinglet — APPROVED** [review]
LGTM

**Pull request merged**